plugin/contrib/mesos/pkg/executor/executor_test.go

```diff
@@ -353,14 +353,21 @@ func TestExecutorLaunchAndKillTask(t *testing.T) {
 		mesosproto.TaskState_TASK_RUNNING,
 	).Return(mesosproto.Status_DRIVER_RUNNING, nil).Run(statusUpdateDone).Once()
 
-	executor.LaunchTask(mockDriver, taskInfo)
-	assert.Equal(t, 1, len(executor.tasks),
-		"executor must be able to create a task")
-
-	// Allow some time for asynchronous requests to the testApiServer.
-	time.Sleep(time.Second)
-	assert.Equal(t, 1, len(executor.pods),
-		"executor must be able to create a pod")
+	func () {
+		executor.LaunchTask(mockDriver, taskInfo)
+
+		// Allow some time for asynchronous requests to the testApiServer. Don't
+		// lock too early because then the executor has no time to take the lock itself.
+		time.Sleep(time.Second)
```

Comment:
Don't like it either. We can do some assert.Eventually style loop.

Let me do that. There are more places in this test file where this would remove arbitrary sleeps.